### ISSUE TITLE
Make product properties writeable by rest-api

### ISSUE THREAD

**Issue by marco-hoyer** [woothemes/woocommerce#7493]
I would like to be able to set the product attributes visible and purchaseable on product creation and update. Actually they're flagged as read-only in the documentation and if I try to set them, woocommerce rest-api ignores them.

**Comment by claudiosmweb** [comment]
There is other things behind it.
This define that the product is purchasable or not:
https://github.com/woothemes/woocommerce/blob/44cf2c2b70fac33b3e8e7c59599e6e2a04d5b41a/includes/abstracts/abstract-wc-product.php#L734-L757

But External and Groups can't be purchasable by default:
https://github.com/woothemes/woocommerce/blob/4e99e629f527c9558a82a31852afe5059bfac48f/includes/class-wc-product-external.php#L31-L39
https://github.com/woothemes/woocommerce/blob/9bc1b09fc5f2efc9c0168b24d6407351a0815385/includes/class-wc-product-grouped.php#L150-L158

In other words, what you want will not be possible.
For Simple and Variable products you can just change the status for draft ;)
